test/ddt_suite/record.lua

```diff
@@ -607,3 +607,114 @@ t {
             {"f1": 13, "f2": null, "f3": null, "f4": null, "f5": 15.0}},
         "f5": 17.0 } ]]
 }
+
+-- nullable xflatten
+local nullable_xflatten = [[{
+    "type": "record", "name": "X", "fields": [
+        {"name": "f1", "type": "int*"},
+        {"name": "f2", "type": {
+            "type": "array*", "items": "string*"}},
+        {"name": "f3", "type":[
+            "int", "X*"]},
+        {"name": "f4", "type": {
+            "type": "map*", "values": "int*"}}]}
+    ]]
+
+t {
+    schema = nullable_xflatten,
+    func = "xflatten",
+    input = '{"f1":null, "f2":null, "f3":{"X":null}, "f4":null}',
```

Comment:
More thoughts. Should we allow to change only leaf nodes of a tuple to prevent a situation when, say, {inner_1 = {inner_2 = 5}} replaced with {inner_1 = {inner_3 = 6}}, but the intention was add the field inner_3 subling to the inner_2? The exception should be explicit NULL… maybe.

Reply:
But now (w/o json path updates) we cannot update any leaf nodes except ones on upper level. Anyway, we need to design the whole functionality now to don’t break it later.

Reply:
>>> box.NULL means 'set to NULL', but omitting a field means 'leave it as is', yep? This is the one place where we treat NULL and omitting differently (I hope), so it should be documented.
1. This behavior was introduced before with nullable unions.
2. Where do you wand this documentation? in README.md?

Reply:
>But now (w/o json path updates) we cannot update any leaf nodes except ones on upper level. Anyway, we need to design the whole functionality now to don’t break it later.
I have created ticket for it #85 with opened discussions.

For now the only possible solution is already implemented. It works like this: in case of complex type updates, xflatten expects the whole object and raises an error if some fields a missing (except for nullable fields)

Reply:
> (except for nullable fields)

So, they will be nullified if were have non-null values before. I think this can be the problem.

Reply:
record* banned in xfaltten for now. A future implementation is under private discussion now, results will be in #85.